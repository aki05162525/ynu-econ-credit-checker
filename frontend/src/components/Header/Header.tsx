export type HeaderProps = {
	title?: string;
};

export const Header = ({ title = "YNU-econ 単位チェッカー" }: HeaderProps) => {
	return (
		<header className="flex items-center justify-center bg-brand p-4 shadow-sm">
			<h1 className="font-semibold text-xl text-white">{title}</h1>
		</header>
	);
};
