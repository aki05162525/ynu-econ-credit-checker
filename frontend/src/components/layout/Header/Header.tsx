import Link from "next/link";

export type HeaderProps = {
	title?: string;
};

export const Header = ({ title = "YNU-econ 単位チェッカー" }: HeaderProps) => {
	return (
		<header className="flex items-center justify-center bg-brand p-4 shadow-sm">
			<Link
				href="/"
				className="text-white hover:text-gray-200 transition-colors"
			>
				<h1 className="font-semibold text-xl">{title}</h1>
			</Link>
		</header>
	);
};
