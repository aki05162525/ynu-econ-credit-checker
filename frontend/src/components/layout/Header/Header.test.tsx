import { render } from "@testing-library/react";
import { describe, expect, test } from "vitest";
import { Header } from "../Header";

describe("Header コンポーネント", () => {
	test("デフォルトのタイトルで表示される", () => {
		const { container } = render(<Header />);
		expect(container).toMatchSnapshot();
	});

	test("カスタムタイトルで表示される", () => {
		const { container } = render(<Header title="単位チェックアプリ" />);
		expect(container).toMatchSnapshot();
	});
});
