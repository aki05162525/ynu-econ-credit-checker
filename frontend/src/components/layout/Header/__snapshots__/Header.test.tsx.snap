// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Header component > renders with custom title 1`] = `
<div>
  <header
    class="flex items-center justify-center bg-brand p-4 shadow-sm"
  >
    <a
      class="text-white hover:text-gray-200 transition-colors"
      href="/"
    >
      <h1
        class="font-semibold text-xl"
      >
        単位チェックアプリ
      </h1>
    </a>
  </header>
</div>
`;

exports[`Header component > renders with default title 1`] = `
<div>
  <header
    class="flex items-center justify-center bg-brand p-4 shadow-sm"
  >
    <a
      class="text-white hover:text-gray-200 transition-colors"
      href="/"
    >
      <h1
        class="font-semibold text-xl"
      >
        YNU-econ 単位チェッカー
      </h1>
    </a>
  </header>
</div>
`;

exports[`Header コンポーネント > renders with custom title 1`] = `
<div>
  <header
    class="flex items-center justify-center bg-brand p-4 shadow-sm"
  >
    <a
      class="text-white hover:text-gray-200 transition-colors"
      href="/"
    >
      <h1
        class="font-semibold text-xl"
      >
        単位チェックアプリ
      </h1>
    </a>
  </header>
</div>
`;

exports[`Header コンポーネント > renders with default title 1`] = `
<div>
  <header
    class="flex items-center justify-center bg-brand p-4 shadow-sm"
  >
    <a
      class="text-white hover:text-gray-200 transition-colors"
      href="/"
    >
      <h1
        class="font-semibold text-xl"
      >
        YNU-econ 単位チェッカー
      </h1>
    </a>
  </header>
</div>
`;

exports[`Header コンポーネント > カスタムタイトルで表示される 1`] = `
<div>
  <header
    class="flex items-center justify-center bg-brand p-4 shadow-sm"
  >
    <a
      class="text-white hover:text-gray-200 transition-colors"
      href="/"
    >
      <h1
        class="font-semibold text-xl"
      >
        単位チェックアプリ
      </h1>
    </a>
  </header>
</div>
`;

exports[`Header コンポーネント > デフォルトのタイトルで表示される 1`] = `
<div>
  <header
    class="flex items-center justify-center bg-brand p-4 shadow-sm"
  >
    <a
      class="text-white hover:text-gray-200 transition-colors"
      href="/"
    >
      <h1
        class="font-semibold text-xl"
      >
        YNU-econ 単位チェッカー
      </h1>
    </a>
  </header>
</div>
`;
